### PR TITLE
Use patched FindCUDA if CMake version 3.7-3.10

### PR DESCRIPTION
**Pull request by karjonas**

### CODE REVIEW

File: ospray_modules/optix/CMakeLists.txt [hunk 1]
```diff
@@ -6,17 +6,17 @@
...
 project(ospray_module_optix VERSION 1.0.0)
 set(ospray_module_optix_VERSION_ABI 1)
 
 include(Common)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+if(CMAKE_VERSION VERSION_GREATER "3.6" AND CMAKE_VERSION VERSION_LESS "3.10")
+    # https://gitlab.kitware.com/cmake/cmake/commit/037beeddd1d434e22040310bc6b88310496d3814#185864a642a40d10362afc05c79aea97451b1058
+    list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/CMakePatch")
+    message(STATUS "Using old CUDA find for bad CMake version")
```
Comment (hernando) [inline ospray_modules/optix/CMakeLists.txt]: Just make it work, don't litter the output

File: ospray_modules/optix/CMakePatch/FindCUDA.cmake [hunk 2]
```diff
@@ -0,0 +1,1819 @@
...
+set(CUDA_FOUND TRUE)
+
+set(CUDA_TOOLKIT_ROOT_DIR_INTERNAL "${CUDA_TOOLKIT_ROOT_DIR}" CACHE INTERNAL
+  "This is the value of the last time CUDA_TOOLKIT_ROOT_DIR was set successfully." FORCE)
+set(CUDA_TOOLKIT_TARGET_DIR_INTERNAL "${CUDA_TOOLKIT_TARGET_DIR}" CACHE INTERNAL
+  "This is the value of the last time CUDA_TOOLKIT_TARGET_DIR was set successfully." FORCE)
+set(CUDA_SDK_ROOT_DIR_INTERNAL "${CUDA_SDK_ROOT_DIR}" CACHE INTERNAL
+  "This is the value of the last time CUDA_SDK_ROOT_DIR was set successfully." FORCE)
+
+include(${CMAKE_CURRENT_LIST_DIR}/FindPackageHandleStandardArgs.cmake)
```
Comment (hernando) [inline ospray_modules/optix/CMakePatch/FindCUDA.cmake]: You don't need to copy FindPackageHandleStandardArgs.cmake in the repo if you replace this include with
include(FindPackageHandleStandardArgs). This way you can also get rid of CMakeParseArguments.cmake and FindPackageMessage.cmake

File: ospray_modules/optix/Context.h [hunk 3]
```diff
@@ -65,6 +65,8 @@ class Context
     Context();
 
     void _initialize();
+    void _printSystemInformation() const;
```
Comment (hernando) [inline ospray_modules/optix/Context.h]: Isn't there any way to control the amount of log output from Brayns. I tend to dislike unwanted logging information, even more when it's for developers.

Comment (karjonas) [reply ospray_modules/optix/Context.h]: You are right, this should be using the BRAYNS_INFO output.

File: ospray_modules/optix/CMakePatch/FindCUDA.cmake (hunk 2 above)
Comment (karjonas) [reply ospray_modules/optix/CMakePatch/FindCUDA.cmake]: That does not seem to work.

**Comment by karjonas** [comment]
retest this please

**Comment by karjonas** [comment]
I have rebased this now and removed some unrelated fixes I will add in another PR.

**Pull request merged**